internal/xds/translator/listener.go

```diff
@@ -340,17 +341,18 @@ func addXdsTLSInspectorFilter(xdsListener *listenerv3.Listener) error {
 	return nil
 }
 
-func buildDownstreamQUICTransportSocket(tlsConfigs []*ir.TLSListenerConfig) (*corev3.TransportSocket, error) {
+func buildDownstreamQUICTransportSocket(tlsConfig *ir.TLSListenerConfig) (*corev3.TransportSocket, error) {
 	tlsCtx := &quicv3.QuicDownstreamTransport{
 		DownstreamTlsContext: &tlsv3.DownstreamTlsContext{
 			CommonTlsContext: &tlsv3.CommonTlsContext{
-				AlpnProtocols: []string{"h3"},
+				TlsParams:     buildTLSParams(tlsConfig),
+				AlpnProtocols: buildALPNProtocols(tlsConfig.ALPNProtocols),
 			},
 			RequireClientCertificate: &wrappers.BoolValue{Value: false},
 		},
 	}
 
-	for _, tlsConfig := range tlsConfigs {
+	for _, tlsConfig := range tlsConfig.Certificates {
```

Comment:
won't these crash if `tlsConfig` is nil ?

Reply:
`tlsConfigs` can't be nil because this function is only called after checking that it isn't.

https://github.com/envoyproxy/gateway/blob/main/internal/xds/translator/listener.go#L190

Reply:
thanks for clarifying this